clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/ClusteredStore.java

```diff
@@ -295,8 +298,22 @@ public void initStore(final Store<?, ?> resource) {
       if (storeConfig == null) {
         throw new IllegalArgumentException("Given store is not managed by this provider : " + resource);
       }
-      ClusteredStore clusteredStore = (ClusteredStore) resource;
-      clusteredStore.storeProxy = clusteringService.getServerStoreProxy(storeConfig.getCacheIdentifier(), storeConfig.getStoreConfig());
+      final ClusteredStore clusteredStore = (ClusteredStore) resource;
+      clusteredStore.storeProxy = clusteringService.getServerStoreProxy(storeConfig.getCacheIdentifier(), storeConfig.getStoreConfig(), storeConfig.getConsistency());
+      clusteredStore.storeProxy.addInvalidationListener(new ServerStoreProxy.InvalidationListener() {
+        @Override
+        public void onInvalidationRequest(long hash) {
+          if (clusteredStore.invalidationValve != null) {
+            try {
+              LOGGER.debug("CLIENT: calling invalidation valve");
+              clusteredStore.invalidationValve.invalidateAllWithHash(hash);
+            } catch (StoreAccessException sae) {
+              //TODO: what should be done here?
```

Comment:
Most likely delegate to the resilience strategy ... so can remain a TODO for now